openprescribing/frontend/tests/commands/test_import_adqs.py

```diff
@@ -1,40 +1,31 @@
+from mock import MagicMock
+from mock import patch
+
 from django.core.management import call_command
 from django.test import TestCase
 from frontend.models import Presentation
 
 
 class CommandsTestCase(TestCase):
+    fixtures = ['presentations', 'importlog']
 
-    def test_import_adqs(self):
-        Presentation.objects.create(bnf_code='0101010J0AAAAAA',
-                                    name='Mag Trisil_Mix')
-        Presentation.objects.create(bnf_code='0104020L0AAAPAP',
-                                    name='Loperamide HCl_Oral Susp 25mg/5ml')
-        Presentation.objects.create(bnf_code='0401010T0AAAEAE',
-                                    name='Temazepam_Oral Soln 10mg/5ml S/F')
-        Presentation.objects.create(bnf_code='0401010T0AAAMAM',
-                                    name='Temazepam_Tab 10mg')
-        Presentation.objects.create(bnf_code='0401020K0AAACAC',
-                                    name='Diazepam_Inj 5mg/ml 2ml Amp')
-
-        args = []
-        opts = {
-            'filename': 'frontend/tests/fixtures/commands/adqs.csv'
-        }
-        call_command('import_adqs', *args, **opts)
-
-        p = Presentation.objects.get(bnf_code='0104020L0AAAPAP')
-        self.assertEqual(p.adq, 8)
-        self.assertEqual(p.adq_unit, 'mg')
-        self.assertEqual(p.active_quantity, 25)
-        self.assertEqual(p.percent_of_adq, 25 / 8.0)
+    @patch('gcutils.bigquery.Client')
+    def test_import_adqs(self, mock_client):
+        mock_query = MagicMock(name='query')
+        mock_query.query.return_value = [
+            ('0202010F0.*AA', 0.333),
+        ]
+        mock_client.return_value = mock_query
+        call_command('import_adqs')
 
-        p = Presentation.objects.get(bnf_code='0401010T0AAAEAE')
-        self.assertEqual(p.adq, 20)
-        self.assertEqual(p.adq_unit, 'mg')
-        self.assertEqual(p.active_quantity, 10)
-        self.assertEqual(p.percent_of_adq, 10 / 20.0)
+        # check we called bigquery with the right source table
+        mock_query.query.assert_called_once()
+        self.assertIn(
+            "tmp_eu.raw_prescribing_data_2014_11",
+            mock_query.query.call_args[0][0])
 
-        p = Presentation.objects.get(bnf_code='0401020K0AAACAC')
-        self.assertEqual(p.adq, None)
-        self.assertEqual(p.adq_unit, None)
+        # Check we set adq_per_quantity according to the bigquery results
+        p = Presentation.objects.get(bnf_code='0202010F0AAAAAA')
+        self.assertEqual(p.adq_per_quantity, 0.333)
+        p = Presentation.objects.get(bnf_code='0204000I0JKKKAL')
+        self.assertEqual(p.adq_per_quantity, None)
```

Comment:
Good to see we're testing both that a change has been made when expected _and_ that a change hasn't been made when not expected.